src/doc_handler.py

```
@@ -0,0 +1,526 @@
+            return ""
+        
+        extracted_text = ""
+        
+        # Method 1: Try pdfplumber first (most accurate)
+        if self.use_pdfplumber:
+            try:
+                extracted_text = self._extract_with_pdfplumber(pdf_bytes)
+                if extracted_text and len(extracted_text.strip()) > 50:
+                    logger.info(f"Successfully extracted {len(extracted_text)} characters using pdfplumber")
+                    return extracted_text
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)